addons/hr_expense/models/hr_expense.py

```diff
@@ -54,24 +54,24 @@ def _get_employee_id_domain(self):
         return res
 
     name = fields.Char('Description', compute='_compute_from_product_id_company_id', store=True, required=True, copy=True,
-        states={'draft': [('readonly', False)], 'reported': [('readonly', False)], 'refused': [('readonly', False)]})
-    date = fields.Date(readonly=True, states={'draft': [('readonly', False)], 'reported': [('readonly', False)], 'refused': [('readonly', False)]}, default=fields.Date.context_today, string="Expense Date")
+        states={'draft': [('readonly', False)], 'reported': [('readonly', False)], 'approved': [('readonly', False)], 'refused': [('readonly', False)]})
```

Comment:
Maybe the `states=` could be reversed :innocent: Put the field as readonly=False and only list the readonly fields in `states`